stdlib/string.gr

```diff
@@ -975,27 +975,44 @@ let writeUtf8CodePoint = (ptr, codePoint) => {
   let (<=) = WasmI32.leU
   let (==) = WasmI32.eq
   if (codePoint <= 0x007Fn) {
+    // Code points in the ASCII range are written as just one byte with the
+    // leading bit equal to zero (0xxxxxxx). Just store the value as one byte
+    // directly. Note that the value is already guaranteed to start with most
+    // significant bit equal to zero because of the check in the if statement
+    // above, so there's no need to bit-mask it.
     WasmI32.store8(ptr, codePoint, 0n)
     1n
   } else if (codePoint <= 0x07FFn) {
-    let high = ((codePoint >>> 6n) & 0x1Fn) | 0xC0n
-    let low = (codePoint & 0x3Fn) | 0x08n
+    // Code points in the range 0x0080..0x07FF are written as two bytes.
+    // The first byte has a three bit prefix of 110, followed by 5 bits of the
+    // codepoint. The second byte has a two bit prefix of 10, followed by 6 bits
+    // of the codepoint.
+		let high = ((codePoint >>> 6n) & 0b000_11111n) | 0b110_00000n
+		let low =  ( codePoint         & 0b00_111111n) | 0b10_000000n
```

Comment:
@ Other reviewers: This appears to be the changed line (there was a typo in the original: `0x08n` should have been `0x80n`)....